### PR TITLE
improve error message on requiring module from the context it was defined in

### PR DESCRIPTION
**Pull request by hubertlepicki**
Issue: https://github.com/elixir-lang/elixir/issues/3881

I think there are two slightly related cases, that may share the same explanation. I tried to make it as clear as possible with wording, but if you do have better idea please let me know and I can amend.

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you @hubertlepicki! I was rather thinking we should leave that message the same and change it only if the context module and the module being checked are the same.

**Comment by hubertlepicki** [comment]
Alright @josevalim, I separated both messages. I am still not sure about the error messages, sort of sounds like mumbo jumbo to me, esp. the first part. that says:

    module Kernel.ErrorsTest.ScheduledModule.Hygiene is not loaded but was defined. 

currently both errors have the same first line like above, they differ in explanation. Maybe you can suggest better wording... you know, this is one of two hard problems in computing...

File: lib/elixir/src/elixir_aliases.erl [hunk 1]
```diff
@@ -141,4 +144,8 @@ format_error({unloaded_module, Module}) ->
 
 format_error({scheduled_module, Module}) ->
   io_lib:format("module ~ts is not loaded but was defined. This happens because you are trying to use a module in the same context it is defined. Try defining the module outside the context that requires it.",
-    [inspect(Module)]).
\ No newline at end of file
+    [inspect(Module)]);
+
+format_error({circular_module, Module}) ->
+  io_lib:format("module ~ts is not loaded but was defined. This happens because you are trying to use a module in the context of the same module. Try using the module from the context of different module.",
```
Comment (josevalim) [inline lib/elixir/src/elixir_aliases.erl]: What about: "you are trying to use/require the same module that is being currently defined. The current module is ~ts"

File: lib/elixir/src/elixir_aliases.erl [hunk 2]
```diff
@@ -81,7 +81,10 @@ ensure_loaded(Meta, Ref, E) ->
   catch
     error:undef ->
       Kind = case lists:member(Ref, ?m(E, context_modules)) of
-        true  -> scheduled_module;
+        true  -> case ?m(E, module) of
```
Comment (lexmag) [inline lib/elixir/src/elixir_aliases.erl]: Would be better to put `case` on the next line, as it's the style throughout the repo.

File: lib/elixir/src/elixir_aliases.erl (hunk 1 above)
Comment (hubertlepicki) [inline lib/elixir/src/elixir_aliases.erl]: :+1:

File: lib/elixir/src/elixir_aliases.erl (hunk 2 above)
Comment (hubertlepicki) [inline lib/elixir/src/elixir_aliases.erl]: right, thanks for the tip

**Comment by hubertlepicki** [comment]
here you go

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**